src/data/thunks.js

```diff
@@ -98,6 +97,32 @@ export function getProctoringSettings() {
   };
 }
 
+// todo: should this happen on its own or within updateAttemptAfter or somewhere else?
```

Comment:
Question: what I'm wondering here is if this function should happen within one of the other functions or if it can be standalone as is. My sense is the latter, as it would wait for the fetch access token call. Plus, I'm not sure that updateAttemptAfter would be the best place for fetchExamAccessTicket anyway since we want to also check for exam access even if there is no exam attempt (for past due exams).

Reply:
I agree that `updateAttemptAfter` is not the best place for this logic to be included. That function is mean to update the state of an attempt, and while the access token logic is related, I wouldn't necessarily say it's tied to the state of an attempt. 

Additionally, `updateAttemptAfter` would be called on every ping (a regular timed interval), which contrasts the original decision to fetch an access token each time new content should be rendered.

Reply:
this makes sense to me as standalone since the trigger for this isn't tied to any of the existing thunks as Alie mentions. We need to kick off this function every time the unit inside xblock iframe url changes. I think you'll have to do some digging around frontend-app-learning to find a good way to accomplish that since as of now this library doesn't know when that changes.